src/test/ReactTestUtils.js

```diff
@@ -500,6 +500,7 @@ function makeSimulator(eventType) {
       fakeNativeEvent,
       node
     );
+    event.persist();
```

Comment:
@syranide 
No. Some tests depend on this issue's behavior.

https://github.com/facebook/react/blob/master/src/test/__tests__/ReactTestUtils-test.js#L433-L465

I prefer that the `SyntheticEvent` in `TestUtils.Simulate` isn't released after the event callback has been invoked.
Should I separate a commit or create an another PR?